### ISSUE TITLE
public_member_api_docs doesn't seem to require docs for mixins

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/linter#2167]
**Describe the issue**
The `public_member_api_docs` lint doesn't seem to warn when a public mixin (or any methods defined in it) do not have a doc comment.

**To Reproduce**
Enable the lint, the following sample doesn't trigger it:

```dart
mixin Foo {
  void foo() {
    // nothing to do.
  }
}
```

**Expected behavior**
Since they are part of the public API, the lint should require doc comments on them.

**Comment by pq** [comment]
Agreed.  This one is showing it's age (predating `mixin`s 😬 ) -- I'll update.  Thanks!

**Comment by bwilkerson** [comment]
We should validate that it works for extensions as well.

**Comment by pq** [comment]
![image](https://user-images.githubusercontent.com/67586/85766602-0dd70380-b6cc-11ea-8713-9611e414ad72.png)

😊
